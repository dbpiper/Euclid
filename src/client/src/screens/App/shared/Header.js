import React from 'react';

import styled from 'styled-components';

import SearchField from './SearchField';

const Head = styled.section`
  text-align: center;
  background: #000000;
  padding-bottom: 1rem;
`;

const Title = styled.span`
  text-align: center;
  margin: 3rem;

  font-size: 3rem;
`;

const SearchArea = styled.div`
  padding-left: 30rem;
`;

const Header = function Header() {
  return (
    <Head>
      <Title>
        Euclid
      </Title>

      <SearchArea>
        <SearchField />
      </SearchArea>
    </Head>
  );
};

export default Header;
