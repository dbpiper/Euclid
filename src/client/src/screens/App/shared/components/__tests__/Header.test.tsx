import { shallow } from 'enzyme';
import React from 'react';
import Header from '../Header';

test('Header has correct title', () => {
  const component = shallow(
    <Header />,
  );

  expect(component.text()).toMatch('Euclid');
  expect(component).toMatchSnapshot();
});
