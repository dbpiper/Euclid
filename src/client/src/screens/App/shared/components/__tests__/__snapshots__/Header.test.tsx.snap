// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header has correct title 1`] = `
<Header__Head>
  <Header__Title>
    Euclid
  </Header__Title>
  <Header__SearchArea>
    <ErrorBoundary
      FallbackComponent={[Function]}
    >
      <Connect(GetTickersSearchField) />
    </ErrorBoundary>
  </Header__SearchArea>
</Header__Head>
`;
